designer/server/src/plugins/routes/api.js

```diff
@@ -0,0 +1,92 @@
+import { Schema } from '@defra/forms-model'
+
+import * as forms from '~/src/lib/forms.js'
+
+/**
+ * @type {ServerRoute}
+ */
+export const getFormWithId = {
+  method: 'GET',
+  path: '/api/{id}/data',
+  options: {
+    async handler(request, h) {
+      const { id } = request.params
+      try {
+        const formJson = await forms.getDraftFormDefinition(id)
+
+        return h.response(/** @type {object} */ (formJson))
+      } catch (error) {
+        request.logger.error(['GET /api/{id}/data', 'getFormWithId'], error)
+        throw error
+      }
+    }
+  }
+}
+
+/**
+ * @type {ServerRoute}
+ */
+export const putFormWithId = {
+  method: 'PUT',
+  path: '/api/{id}/data',
+  options: {
+    payload: {
+      parse: true
+    },
+    async handler(request, h) {
+      const { id } = request.params
+      try {
+        const { value, error } = Schema.validate(request.payload, {
+          abortEarly: false
+        })
+
+        if (error) {
+          request.logger.error(
+            ['error', `/api/${id}/data`],
+            [error, request.payload]
+          )
+
+          throw new Error(`Schema validation failed, reason: ${error.message}`)
+        }
+
+        await forms.updateDraftFormDefinition(id, JSON.stringify(value))
```

Comment:
Don't think we need to stringify this ([see comment](https://github.com/DEFRA/forms-designer/pull/76/files#r1579793458))

```suggestion
        await forms.updateDraftFormDefinition(id, value)
```